src/adapters/xhb.js

```diff
@@ -0,0 +1,187 @@
+
+var XhbAdapter = function XhbAdapter() {
+
+    function buildJPTCall(bid, callbackId) {
+        //determine tag params
+        var placementId = utils.getBidIdParamater('placementId', bid.params);
+        var inventoryCode = utils.getBidIdParamater('invCode', bid.params);
+        var referrer = utils.getBidIdParamater('referrer', bid.params);
+        var altReferrer = utils.getBidIdParamater('alt_referrer', bid.params);
+
+        //Always use https
+        var jptCall = 'https://ib.adnxs.com/jpt?';
+
+        jptCall = utils.tryAppendQueryString(jptCall, 'callback', '$$PREBID_GLOBAL$$.handleXhbCB');
+        jptCall = utils.tryAppendQueryString(jptCall, 'callback_uid', callbackId);
+        jptCall = utils.tryAppendQueryString(jptCall, 'id', placementId);
+        jptCall = utils.tryAppendQueryString(jptCall, 'code', inventoryCode);
+
+        //sizes takes a bit more logic
+        var sizeQueryString = '';
+        var parsedSizes = utils.parseSizesInput(bid.sizes);
+
+        //combine string into proper querystring for impbus
+        var parsedSizesLength = parsedSizes.length;
+        if (parsedSizesLength > 0) {
+            //first value should be "size"
+            sizeQueryString = 'size=' + parsedSizes[0];
+            if (parsedSizesLength > 1) {
+                //any subsequent values should be "promo_sizes"
+                sizeQueryString += '&promo_sizes=';
+                for (var j = 1; j < parsedSizesLength; j++) {
+                    sizeQueryString += parsedSizes[j] += ',';
+                }
+                //remove trailing comma
+                if (sizeQueryString && sizeQueryString.charAt(sizeQueryString.length - 1) === ',') {
+                    sizeQueryString = sizeQueryString.slice(0, sizeQueryString.length - 1);
+                }
+            }
+        }
+
+        if (sizeQueryString) {
+            jptCall += sizeQueryString + '&';
+        }
+
+
+
+
+        //append custom attributes:
+        var paramsCopy = utils.extend({}, bid.params);
+
+        //delete attributes already used
+        delete paramsCopy.placementId;
+        delete paramsCopy.invCode;
+        delete paramsCopy.query;
+        delete paramsCopy.referrer;
+        delete paramsCopy.alt_referrer;
+
+        //get the reminder
+        var queryParams = utils.parseQueryStringParameters(paramsCopy);
+
+        //append
+        if (queryParams) {
+            jptCall += queryParams;
+        }
+
+        //append referrer
+        if (referrer === '') {
+            referrer = utils.getTopWindowUrl();
+        }
+
+        jptCall = utils.tryAppendQueryString(jptCall, 'referrer', referrer);
+        jptCall = utils.tryAppendQueryString(jptCall, 'alt_referrer', altReferrer);
+
+
+
+
+        //remove the trailing "&"
+        if (jptCall.lastIndexOf('&') === jptCall.length - 1) {
+            jptCall = jptCall.substring(0, jptCall.length - 1);
+        }
+
+        // @if NODE_ENV='debug'
```

Comment:
these code comments don't actually work anymore. You can strip out any blocks with this syntax.